docs/topics/tour/kotlin-tour-intermediate-classes-interfaces.md

```diff
@@ -295,20 +295,21 @@ class Circle : Drawable {
     override fun draw() {
         TODO("An example implementation")
     }
-    
+
     override fun resize() {
         TODO("An example implementation")
     }
-   override val color = null
+
+    override val color = null
 }
 ```
 
-If you wanted to create a child class of the `Circle` class which had the same behavior **except** for the value of the
-`color` property, you still need to add implementations for each member function of the `Circle` class:
+If you wanted to create a new class which implements the `Drawable` interface, and have the same behavior with `Circle`
+class **except** for the value of the `color` property, you still need to add implementations for each member function
+of the `Drawable` interface:
 
 ```kotlin
-class RedCircle(val circle: Circle) : Circle {
-
+class RedCircle(val circle: Circle) : Drawable {
```

Comment:
JFYI
IMO:
if class `RedCircle` extends class `Circle`, it will inherit all the member functions, which means there will be no necessary to implement func `draw()` and `resize()`

so I think the sample code here will confuse the readers, and I suggest to change it

Reply:
You're right that this example could be improved. I'm covering this in https://github.com/JetBrains/kotlin-web-site/pull/5026 Can you revert the changes in this file since I'll replace it very soon?

Reply:
OK
I reverted this change